src/tests/TouchFree.test.ts

```diff
@@ -78,9 +79,68 @@ describe('TouchFree', () => {
         expect(TouchFree.GetCurrentCursor()).toBe(newCursor);
     });
 
-    it('GetInputController should get the input controller correctly', () => {
+    test('GetInputController should get the input controller correctly', () => {
         TouchFree.Init();
         const controller = TouchFree.GetInputController();
         expect(controller instanceof WebInputController).toBe(true);
     });
+
+    test('ControlAnalyticsSession should call AnalyticsSessionRequest with the correct arguments', () => {
+        ConnectionManager.init();
+        const serviceConnection = ConnectionManager.serviceConnection();
+        if (!serviceConnection) fail('Service connection not available');
+
+        const applicationName = 'testApplication';
+
+        const testFn = jest
+            .spyOn(serviceConnection, 'AnalyticsSessionRequest')
+            .mockImplementation((requestType, sessionID, callback) => {
+                expect(sessionID.includes(applicationName)).toBe(true);
+                callback?.(new WebSocketResponse('test', 'Success', 'test', 'test'));
+                return requestType;
+            });
+
+        TouchFree.ControlAnalytics('START', applicationName);
+        expect(testFn).toReturnWith('START');
+
+        TouchFree.ControlAnalytics('STOP', applicationName);
+        expect(testFn).toReturnWith('STOP');
+    });
+
+    test('ControlAnalyticsSession should give appropriate warnings on START', () => {
+        ConnectionManager.init();
+        const serviceConnection = ConnectionManager.serviceConnection();
+        if (!serviceConnection) fail('Service connection not available');
+
+        const applicationName = 'testApplication';
+
+        let id = '';
+
+        jest.spyOn(serviceConnection, 'AnalyticsSessionRequest').mockImplementation((_arg1, sessionID, callback) => {
+            id = sessionID;
+            callback?.(new WebSocketResponse('test', 'Success', 'test', 'test'));
+        });
+        jest.spyOn(console, 'warn').mockImplementation((arg) => {
+            expect(arg).toBe(`Session: ${id} already in progress`);
+        });
+
+        TouchFree.ControlAnalytics('START', applicationName);
+        TouchFree.ControlAnalytics('START', applicationName);
+        TouchFree.ControlAnalytics('STOP', applicationName);
+    });
+
+    test('ControlAnalyticsSession should give appropriate warnings on STOP', () => {
+        ConnectionManager.init();
+        const serviceConnection = ConnectionManager.serviceConnection();
+        if (!serviceConnection) fail('Service connection not available');
+
+        const applicationName = 'testApplication';
+
+        jest.spyOn(serviceConnection, 'AnalyticsSessionRequest').mockImplementation(() => {});
+        jest.spyOn(console, 'warn').mockImplementation((arg) => {
+            expect(arg).toBe('No active session');
```

Comment:
💬 Maybe add a check to ensure this is called? As the test will pass if console.warn isn't called